sympy/combinatorics/permutations.py

```diff
@@ -2995,13 +2995,14 @@ def __new__(cls, perm, x, evaluate=None):
         if evaluate is None:
             evaluate = global_parameters.evaluate
 
-        perm = _sympify(perm)
-        x = _sympify(x)
-
-        if not isinstance(perm, Permutation):
+        try:
+            perm = _sympify(perm, Permutation)
+        except NotImplementedError:
```

Comment:
If possible, can we replace try-except block with an if-else check?

Reply:
I don't think try_except can be replaced with if-else in this case. You can suggest if you know any, I am to suggestions.